beater/beater.go

```diff
@@ -9,7 +9,7 @@ import (
 )
 
 type beater struct {
-	done   chan struct{}
+	errCh  chan error
```

Comment:
By "not only an error" I was not referring to the message coming from the Listener, but the way the `done` channel is used. If the channel contains an error I'm definitively ok with the name, but I think we don't even need to send an error over the channel if the code is changed.